src/t_cose_crypto.h

```diff
@@ -154,6 +156,137 @@ t_cose_crypto_sig_size(int32_t            cose_algorithm_id,
                        struct t_cose_key  signing_key,
                        size_t            *sig_size);
 
+/**
+ * \brief Returns the requested number of random bytes.
+ *
+ * \param[in] buffer             Pointer and length of buffer into which
+ *                               the resulting random bytes are put.
+ *
+ * This function will either return the requested number of random bytes,
+ * or produce an error.
+ *
+ * \retval T_COSE_SUCCESS
+ *         Successfully returned the requested number of random bytes.
+ * \retval T_COSE_ERR_RNG_FAILED
+ *         The random number generator failed to return the requested
+ *         number of bytes.
+ */
+enum t_cose_err_t
+t_cose_crypto_get_random(struct q_useful_buf buffer);
+
```

Comment:
There should be an additional out parameter that is a struct q_useful_buf_c * where const output is returned. Yes, this is entirely redundant in that the pointer and length will be the same, but this is the way usefulbufs work. Non-const buffer to write to [in], const buffer with data [out].

It will also clean up the calling code where you make the nonce. You won't need the cast to const there.